doc/library-maintainer.rst

```diff
@@ -0,0 +1,307 @@
+=======================================
+Maintaining Neptune-Versioned Libraries
+=======================================
+
+This is a guide for maintainers of libraries which are using a strict, Neptune-
+based versioning scheme. After some discussion, it provides a set of simple,
+step-by-step guides describing the common procedures for various actions that
+you'll need to perform as a maintainer of a Neptune-versioned library.
+
+You should first read `Using Neptune-Versioned Libraries <neptune-user.rst>`_ to
+get a grounding in the principles and theory of Neptune from the point-of-view
+of a user of a library.
```

Comment:
It might be good to add a some point a more general (and brief) overview about this project, and link to it. See #2 and #3.